src/dynamics/initial_conditions.jl

```diff
@@ -112,6 +112,72 @@ function initialize!(   progn::PrognosticVariables{NF},
     set!(progn, model; vor=ξ, lf=1)
 end
 
+"""$(TYPEDSIGNATURES)
+Kernel version of initialize! for RandomVorticity initial conditions.
+Uses GPU-compatible operations for parallel execution."""
+function initialize!(
+    progn::PrognosticVariables{NF},
+    initial_conditions::RandomVorticity,
+    model::Barotropic
+) where NF
+
+    # reseed the random number generator, for seed=0 randomly seed from Julia's global RNG
+    seed = initial_conditions.seed == 0 ? rand(UInt) : initial_conditions.seed
+    RNG = initial_conditions.random_number_generator
+    Random.seed!(RNG, seed)
+
+    lmax = progn.trunc + 1
+    power = initial_conditions.power + 1    # +1 as power is summed of orders m
+
+    (; spectrum, nlayers) = progn
+    A = convert(NF, initial_conditions.amplitude)
+    
+    # Pre-generate random values on CPU, then transfer to device
+    nlm = length(spectrum)
+    random_values_cpu = 2 .* rand(RNG, Complex{NF}, nlm, nlayers) .- (1 + 1im)
```

Comment:
As long as we don't create arrays of random values during model integration we just do it on the CPU and transfer right? Like the GPUs all have some `rand` methods too but not it's worth it ...